cmd/influxd/upgrade/upgrade.go

```diff
@@ -1,41 +1,80 @@
+//go:generate env GO111MODULE=on go run github.com/kevinburke/go-bindata/go-bindata -o upgrade_gen.go -ignore go -pkg upgrade .
+
 package upgrade
 
 import (
 	"context"
+	"errors"
 	"fmt"
 	"os"
 	"os/user"
 	"path/filepath"
+	"runtime"
+	"time"
 
 	"github.com/influxdata/influxdb/v2"
-	"github.com/influxdata/influxdb/v2/authorizer"
 	"github.com/influxdata/influxdb/v2/bolt"
 	"github.com/influxdata/influxdb/v2/dbrp"
 	"github.com/influxdata/influxdb/v2/internal/fs"
+	"github.com/influxdata/influxdb/v2/kit/cli"
 	"github.com/influxdata/influxdb/v2/kit/metric"
 	"github.com/influxdata/influxdb/v2/kit/prom"
 	"github.com/influxdata/influxdb/v2/kv"
 	"github.com/influxdata/influxdb/v2/kv/migration"
 	"github.com/influxdata/influxdb/v2/kv/migration/all"
+	fs2 "github.com/influxdata/influxdb/v2/pkg/fs"
+	"github.com/influxdata/influxdb/v2/storage"
+	options2 "github.com/influxdata/influxdb/v2/task/options"
 	"github.com/influxdata/influxdb/v2/tenant"
 	"github.com/influxdata/influxdb/v2/v1/services/meta"
 	"github.com/influxdata/influxdb/v2/v1/services/meta/filestore"
 	"github.com/spf13/cobra"
 	"go.uber.org/zap"
 )
 
-var Command = &cobra.Command{
-	Use:   "upgrade",
-	Short: "Upgrade a 1.x version of InfluxDB",
-	RunE:  runUpgradeE,
+// Simplified 1.x config.
+type configV1 struct {
+	Meta struct {
+		Dir string `toml:"dir"`
+	} `toml:"meta"`
+	Data struct {
+		Dir    string `toml:"dir"`
+		WALDir string `toml:"wal-dir"`
+	} `toml:"data"`
 }
 
 type optionsV1 struct {
 	metaDir string
+	walDir  string
+	dataDir string
+	// cmd option
+	dbDir      string
+	configFile string
+}
+
+func (o *optionsV1) checkDirs() error {
+	if o.metaDir == "" || o.dataDir == "" || o.walDir == "" {
+		if o.dbDir == "" {
+			return errors.New("source directory not specified")
+		} else {
+			o.metaDir = filepath.Join(o.dbDir, "meta")
+			o.dataDir = filepath.Join(o.dbDir, "data")
+			o.walDir = filepath.Join(o.dbDir, "wal")
+		}
+	}
+	return nil
 }
 
 type optionsV2 struct {
-	boltPath string
+	boltPath           string
+	enginePath         string
+	userName           string
+	password           string
+	orgName            string
+	bucket             string
+	token              string
+	retention          string
```

Comment:
```suggestion
	retention          time.Duration
```
Then you won't have to go to the trouble of parsing it yourself.